src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -184,11 +192,18 @@ function _is_resource_fit(arn_path, statement) {
         dbg.log1('bucket_policy: ', statement.Resource ? 'Resource' : 'NotResource', ' fit?', resource_regex, arn_path);
         if (resource_regex.test(arn_path)) {
             resource_fit = true;
+            break;
         }
     }
     return statement.Resource ? resource_fit : !resource_fit;
 }
 
+async function is_statement_fit_of_method_array(statements, account, method_arr, arn_path, req) {
+    const res_arr = await Promise.all(method_arr.map(method_permission =>
+        _is_statements_fit(statements, account, method_permission, arn_path, req)));
+    return res_arr;
```

Comment:
Nitpick: Just `return Promise.all(method_arr.map(method_permission => _is_statements_fit(statements, account, method_permission, arn_path, req)))` would be sufficient.

Reply:
I will change it, thanks.